### ISSUE TITLE
[BUG] update_predict in `ReducedRegressionForecaster` does not update the parameter values based on data in sliding/expanding windows

### ISSUE THREAD

**Issue by ngupta23** [alan-turing-institute/sktime#621]
**Describe the bug**
When performing `update_predict()` on `ReducedRegressionForecaster`, the parameter values are not updated using the latest data. 

**To Reproduce**

I will add some code later if needed, but here is the code stack:

[`update_predict`](https://github.com/alan-turing-institute/sktime/blob/v0.5.2/sktime/forecasting/base/_sktime.py#L580)  --> [`_predict_moving_cutoff`](https://github.com/alan-turing-institute/sktime/blob/v0.5.2/sktime/forecasting/base/_sktime.py#L610)  --> [`_update_predict_single`](https://github.com/alan-turing-institute/sktime/blob/36f3cb1b96421bb08e041d16949d3d8a9be6f7d8/sktime/forecasting/base/_sktime.py#L456) --> This does a `self.update()`, but it simply updates the data values using `_update_y_X()` and does not refit the model with this data to update the parameters.

**Expected behavior**
After each update of the data, one would expect that the model parameters also be updated using the latest data. 

**Additional context**
Similar issue can be found in https://github.com/alan-turing-institute/sktime/issues/431

**Versions**
<details>

```python
System:
    python: 3.6.12 |Anaconda, Inc.| (default, Sep  9 2020, 00:29:25) [MSC v.1916 64 bit (AMD64)]
executable: C:\Users\Nikhil\.conda\envs\sktime_dev\python.exe
   machine: Windows-10-10.0.18362-SP0

Python dependencies:
          pip: 20.3.3
   setuptools: 51.0.0.post20201207
      sklearn: 0.24.0
       sktime: 0.5.1
  statsmodels: 0.12.1
        numpy: 1.19.4
        scipy: 1.5.4
       Cython: 0.29.17
       pandas: 1.1.5
   matplotlib: 3.3.3
       joblib: 1.0.0
        numba: 0.52.0
     pmdarima: 1.8.0
      tsfresh: 0.17.0
```

</details>

**Comment by fkiraly** [comment]
@ngupta23, I don't think this is a bug, but just a lack of specification what update should do, and implementation.

I would suggest you add a parameter (at constuction) that allows to specify updating behaviour, either of:
* does not update parameters (current behaviour) - default behaviour i.m.o., since least "invasive"
* updates parameters at each `update` call, using a lookback batch with fixed lookback window
* updates parameters at each `update`, using "full lookback"
* uses `update` method of the wrapped estimator (requires the wrapped estimator to have `update`)

**Comment by mloning** [comment]
We currently have a `update_params` as a keyword argument for the `update` method but nothing has been implemented yet, part of the reason is that updating like fitting is algorithm specific.

**Comment by ngupta23** [comment]
@fkiraly Thanks for the feedback. I like the idea of adding the argument to control the functionality. That has 2 advantages

* Forces documentation of the inner functionality, even if some of the options are not implemented.
* Once the options are implemented, it provides flexibility to the end user.

**Comment by mloning** [comment]
Closed by #660
